### PR TITLE
[AntiForgery]Add docs and update names for readability

### LINKED ISSUE #1856: Refactor anti-forgery code
Antiforgery code needs to be refactored.As part of this we should use better variable and method names and make it more readable.

### PR DESCRIPTION
**Pull request by kirthik**
Fix #1856 
I adeed docs for public apis and updated some variable names.

@harshgMSFT : Could you please take a look?

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/AntiForgery/AntiForgeryTokenSet.cs [hunk 1]
```diff
@@ -22,11 +30,19 @@ public AntiForgeryTokenSet(string formToken, string cookieToken)
...
+        /// </summary>
         public string FormToken { get; private set; }
 
-        // The cookie token is allowed to be null.
-        // This would be the case when the old cookie token is still valid.
-        // In such cases a call to GetTokens would return a token set with null cookie token.
+        /// <summary>
+        /// The token that is supplied in the request cookie.
+        /// </summary>
+        /// <remarks>
```
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/AntiForgery/AntiForgeryTokenSet.cs]: leave these as comments.

File: test/Microsoft.AspNet.Mvc.Core.Test/AntiXsrf/ITokenProvider.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 namespace Microsoft.AspNet.Mvc.Core.Test
 {
     // A TokenProvider that can be passed to MoQ
-    internal interface ITokenProvider : ITokenValidator, ITokenGenerator
+    internal interface ITokenProvider : IAntiForgeryTokenValidator, IAntiForgeryTokenGenerator
```
Comment (harshgMSFT) [inline test/Microsoft.AspNet.Mvc.Core.Test/AntiXsrf/ITokenProvider.cs]: why not `IAntiForgeryTokenProvider` ? rename the file as well.

File: test/Microsoft.AspNet.Mvc.Core.Test/AntiXsrf/AntiForgeryWorkerTest.cs [hunk 3]
```diff
@@ -459,11 +459,11 @@ private Mock<ITokenProvider> GetTokenProvider(HttpContext context, TestTokenSet
             return mockValidator;
         }
 
-        private Mock<ITokenStore> GetTokenStore(HttpContext context, TestTokenSet testTokenSet, bool saveNewCookie = true)
+        private Mock<IAntiForgeryTokenStore> GetTokenStore(HttpContext context, TestTokenSet testTokenSet, bool saveNewCookie = true)
```
Comment (harshgMSFT) [inline test/Microsoft.AspNet.Mvc.Core.Test/AntiXsrf/AntiForgeryWorkerTest.cs]: line break.

File: src/Microsoft.AspNet.Mvc.Core/AntiForgery/IClaimUidExtractor.cs [hunk 4]
```diff
@@ -5,9 +5,16 @@
...
 namespace Microsoft.AspNet.Mvc
 {
-    // Can extract unique identifers for a claims-based identity
+    /// <summary>
+    /// This interface can extract unique identifers for a claims-based identity.
+    /// </summary>
     public interface IClaimUidExtractor
     {
+        /// <summary>
+        /// Extracts claims Uid.
```
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/AntiForgery/IClaimUidExtractor.cs]: what is claims Uid ?

File: src/Microsoft.AspNet.Mvc.Core/AntiForgery/IClaimUidExtractor.cs [hunk 5]
```diff
@@ -5,9 +5,16 @@
...
+        /// </summary>
+        /// <param name="identity">The <see cref="ClaimsIdentity"/>.</param>
+        /// <returns>The claims uid.</returns>
```
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/AntiForgery/IClaimUidExtractor.cs]: what is claims Uid ?

File: src/Microsoft.AspNet.Mvc.Core/AntiForgery/AntiForgeryWorker.cs [hunk 6]
```diff
@@ -132,25 +132,25 @@ public TagBuilder GetFormInputElement([NotNull] HttpContext httpContext)
         public AntiForgeryTokenSet GetTokens([NotNull] HttpContext httpContext, string serializedOldCookieToken)
```
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/AntiForgery/AntiForgeryWorker.cs]: `serializedOldCookieToken` -> `cookieToken`

**Comment by harshgMSFT** [comment]
:shipit:  after fixing minor comments.

**Comment by kirthik** [comment]
Addressed all comments above. Code checked in - 3c9456e2a33ce0d5a518ebad59b399421b96f87b

**Pull request closed without merging**